File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_agedcreditoroutstanding.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_agedcurrencyoutstanding as shared_codat_datacontracts_datasets_agedcurrencyoutstanding
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsAgedCreditorOutstanding:
    aged_currency_outstanding: Optional[list[shared_codat_datacontracts_datasets_agedcurrencyoutstanding.CodatDataContractsDatasetsAgedCurrencyOutstanding]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('agedCurrencyOutstanding'), 'exclude': lambda f: f is None }})
    supplier_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('supplierId'), 'exclude': lambda f: f is None }})
    supplier_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('supplierName'), 'exclude': lambda f: f is None }})
    

File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsagedcreditoroutstanding.go
package shared

type CodatDataContractsDatasetsAgedCreditorOutstanding struct {
	AgedCurrencyOutstanding []CodatDataContractsDatasetsAgedCurrencyOutstanding `json:"agedCurrencyOutstanding,omitempty"`
	SupplierID              *string                                             `json:"supplierId,omitempty"`
	SupplierName            *string                                             `json:"supplierName,omitempty"`
}
